### ISSUE TITLE
Use 'allowJs' with 'declaration' (remove TS5053)

### ISSUE THREAD

**Issue by mihailik** [Microsoft/TypeScript#16061]
**TypeScript Version:**  2.3.2

Lots of libraries have JSDoc comments, and wish to generate `.d.ts` files out of them. TS already has machinery to harvest and check types off JS files. Unfortunately, currently declarations are explicitly blocked by TS5053 error.

Build with `--allowJS --declaration` arguments.

```js
/**
 * @param {string} input
 */
function foo(input) {
}
```

**Expected behavior:**
To build fine and to produce `.d.ts` something like:
```ts
function foo(input: string);
```

**Actual behavior:**
```
error TS5053: Option 'allowJs' cannot be specified with option 'declaration'.
```

**Comment by mihailik** [comment]
An example of such request from [preact.js library](https://github.com/developit/preact/issues/679):

## [Use typescript's --checkJS to validate preact doc comments](https://github.com/developit/preact/issues/679)

> @robertknight
![preact-allowjs-issue](https://cloud.githubusercontent.com/assets/4041967/26414741/f782c436-40a7-11e7-9254-d2c7faa902e2.png)

**Comment by mhegazy** [comment]
![image](https://cloud.githubusercontent.com/assets/8000722/26415067/0cc59068-4066-11e7-8cf6-03d07d618e4a.png)

**Comment by mhegazy** [comment]
Duplicate of https://github.com/Microsoft/TypeScript/issues/7546
